run_test.go

```diff
@@ -157,3 +157,29 @@ func TestRunShmSize(t *testing.T) {
 
 	base.Cmd("run", "--rm", "--shm-size", shmSize, testutil.AlpineImage, "/bin/grep", "shm", "/proc/self/mounts").AssertOutContains("size=32768k")
 }
+
+func TestRunEnvFile(t *testing.T) {
+	base := testutil.NewBase(t)
+
+	const pattern = "env-file"
+	file1, err := ioutil.TempFile("", pattern)
+	assert.NilError(base.T, err)
+	path1 := file1.Name()
+	defer file1.Close()
+	defer os.Remove(path1)
+	err = ioutil.WriteFile(path1, []byte("# this is a comment line\nTESTKEY1=TESTVAL1"), 0666)
+	assert.NilError(base.T, err)
+
+	file2, err := ioutil.TempFile("", pattern)
+	assert.NilError(base.T, err)
+	path2 := file2.Name()
+	defer file2.Close()
+	defer os.Remove(path2)
+	err = ioutil.WriteFile(path2, []byte("# this is a comment line\nTESTKEY2=TESTVAL2"), 0666)
+	assert.NilError(base.T, err)
+
+	base.Cmd("run", "--rm", "--env-file", path1, "--env-file", path2, testutil.AlpineImage,
+		"/usr/bin/env", "|", "grep", "TESTKEY1", "|", "awk", "-F=", "'{print $2}'").AssertOutContains("TESTVAL1")
```

Comment:
Why not just echo $TESTKEY1

Reply:
Sry for the mistake I made, but here I found when I use echo as init command, the output env var is not correct, same thing happens using `env` flag like:
`nerdctl --namespace=nerdctl-test run --rm -e TESTKEY=TESTVALUE ghcr.io/stargz-containers/alpine:3.13-org echo $TESTKEY`
Output should be `TESTVALUE` but actually nothing. And when I attached to the container and run `echo $TESTKEY`, it seems working. (this also happens with `docker run`). And then I found host env will be outputed when echo is init command by `echo $PATH`.

Is it normal? Or something wrong with my host's env or config?

Reply:
Probably you forgot to quote `'$TESTKEY'`

Reply:
> Probably you forgot to quote `'$TESTKEY'`

quote doesn't work, instead `$TESTKEY` will be the output

Reply:
"sh" , "-c", "echo '$TESTKEY'"

Reply:
you should escape  the $ with `\` in the runner shell .
Try : 
`nerdctl --namespace=nerdctl-test run --rm -e TESTKEY=TESTVALUE ghcr.io/stargz-containers/alpine:3.13-org sh -c " echo \$TESTKEY"`

Reply:
> you should escape the $ with `\` in the runner shell .

thx a lot, that works

Reply:
it looks like test pass without escaping the $.  How did you manage to fix it ?